x/ibc/04-channel/types/msgs_test.go

```diff
@@ -303,3 +306,130 @@ func TestMsgChannelCloseConfirm(t *testing.T) {
 		}
 	}
 }
+
+var _ PacketDataI = validPacketT{}
+
+type validPacketT struct{}
+
+func (validPacketT) GetCommitment() []byte {
+	return []byte("testdata")
+}
+
+func (validPacketT) GetTimeoutHeight() uint64 {
+	return 100
+}
+
+func (validPacketT) ValidateBasic() sdk.Error {
+	return nil
+}
+
+func (validPacketT) Type() string {
+	return "valid"
+}
+
+var _ PacketDataI = invalidPacketT{}
+
+type invalidPacketT struct{}
+
+func (invalidPacketT) GetCommitment() []byte {
```

Comment:
Commitments (e.g. the choice of hash function) should be specified in the IBC keeper, not per-packet, I think. Why is this here?

Reply:
I misunderstood the specs, will fix it

Reply:
Has this been resolved in the latest work @mossid ?

Reply:
Still needs to be fixed, as far as I can tell (at least in this file).

Reply:
This has been fixed in the logic, the IBC channel keeper calls `channeltypes.CommitPacket(uint64, PacketDataI)` to hash the timeout height with the commitment bytes. Will remove the mock packet type per @fedekunze 's comment